<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Gestor de Inventario IoT</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="groupBoxSearch">
      <property name="title">
       <string>Búsqueda</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayoutSearch">
       <item>
        <widget class="QLabel" name="labelSearch">
         <property name="text">
          <string>Buscar:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="searchEdit">
         <property name="placeholderText">
          <string>Nombre, tipo o ubicación...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="checkStockButton">
         <property name="text">
          <string>Verificar Stock Bajo</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTableView" name="tableView">
      <property name="alternatingRowColors">
       <bool>true</bool>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectRows</enum>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::SingleSelection</enum>
      </property>
      <property name="sortingEnabled">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBoxForm">
      <property name="title">
       <string>Agregar/Editar Componente</string>
      </property>
      <layout class="QGridLayout" name="gridLayoutForm">
       <item row="0" column="0">
        <widget class="QLabel" name="labelName">
         <property name="text">
          <string>Nombre:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="nameEdit">
         <property name="placeholderText">
          <string>Ej: Resistor 10kΩ</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QLabel" name="labelType">
         <property name="text">
          <string>Tipo:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QLineEdit" name="typeEdit">
         <property name="placeholderText">
          <string>Ej: Resistor, Capacitor, Placa</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="labelQuantity">
         <property name="text">
          <string>Cantidad:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="quantitySpin">
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>9999</number>
         </property>
         <property name="value">
          <number>1</number>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QLabel" name="labelLocation">
         <property name="text">
          <string>Ubicación:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QLineEdit" name="locationEdit">
         <property name="placeholderText">
          <string>Ej: Estante A1, Laboratorio</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="labelDate">
         <property name="text">
          <string>Fecha de Compra:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QDateEdit" name="dateEdit">
         <property name="calendarPopup">
          <bool>true</bool>
         </property>
         <property name="displayFormat">
          <string>dd/MM/yyyy</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2" colspan="2">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayoutButtons">
      <item>
       <widget class="QPushButton" name="addButton">
        <property name="text">
         <string>Agregar</string>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: #4CAF50; color: white; padding: 8px; font-weight: bold;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="updateButton">
        <property name="text">
         <string>Actualizar</string>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: #2196F3; color: white; padding: 8px; font-weight: bold;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="deleteButton">
        <property name="text">
         <string>Eliminar</string>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: #F44336; color: white; padding: 8px; font-weight: bold;</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacerButtons">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="clearButton">
        <property name="text">
         <string>Limpiar Formulario</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>900</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuArchivo">
    <property name="title">
     <string>Archivo</string>
    </property>
    <addaction name="actionSalir"/>
   </widget>
   <widget class="QMenu" name="menuHerramientas">
    <property name="title">
     <string>Herramientas</string>
    </property>
    <addaction name="actionVerificar_Stock"/>
    <addaction name="separator"/>
    <addaction name="actionBackup_BD"/>
    <addaction name="actionRestaurar_BD"/>
   </widget>
   <widget class="QMenu" name="menuAyuda">
    <property name="title">
     <string>Ayuda</string>
    </property>
    <addaction name="actionAcerca_de"/>
   </widget>
   <addaction name="menuArchivo"/>
   <addaction name="menuHerramientas"/>
   <addaction name="menuAyuda"/>
  </widget>
  <action name="actionSalir">
   <property name="text">
    <string>Salir</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionVerificar_Stock">
   <property name="text">
    <string>Verificar Stock Bajo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionBackup_BD">
   <property name="text">
    <string>Backup Base de Datos</string>
   </property>
  </action>
  <action name="actionRestaurar_BD">
   <property name="text">
    <string>Restaurar Base de Datos</string>
   </property>
  </action>
  <action name="actionAcerca_de">
   <property name="text">
    <string>Acerca de...</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
